src/main/java/it/gov/pagopa/pu/organization/service/organization/OrganizationService.java

```diff
@@ -0,0 +1,32 @@
+package it.gov.pagopa.pu.organization.service.organization;
+
+import it.gov.pagopa.pu.organization.dto.generated.OrganizationApiKeys;
+import it.gov.pagopa.pu.organization.exception.custom.OrganizationNotFoundException;
+import it.gov.pagopa.pu.organization.repository.OrganizationRepository;
+import jakarta.transaction.Transactional;
+import org.springframework.stereotype.Service;
+
+@Service
+public class OrganizationService {
+
+  private final OrganizationEncryptionService organizationEncryptionService;
+  private final OrganizationRepository organizationRepository;
+
+  public OrganizationService(OrganizationEncryptionService organizationEncryptionService, OrganizationRepository organizationRepository) {
+    this.organizationEncryptionService = organizationEncryptionService;
+    this.organizationRepository = organizationRepository;
+  }
+
+  @Transactional
+  public void encryptAndSaveApiKey(Long organizationId, OrganizationApiKeys organizationApiKeys){
+    if (!organizationRepository.existsById(organizationId)) {
+      throw new OrganizationNotFoundException(
+        "Organization with id %s was not found".formatted(organizationId)
+      );
+    }
```

Comment:
it's not necessary to search the organization... if the update give not result, it has not found the organization!